File: php/features.php
<?php
declare(strict_types=1);

// Mediazona SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MediazonaFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MediazonaBaseFeature();
            case "test":
                return new MediazonaTestFeature();
            default:
                return new MediazonaBaseFeature();
        }
    }
}

File: py/features.py
# Mediazona SDK feature factory

from feature.base_feature import MediazonaBaseFeature
from feature.test_feature import MediazonaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MediazonaBaseFeature(),
        "test": lambda: MediazonaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
